frontend/packages/git-service/src/utils/pac-strategy-detector.ts

```diff
@@ -0,0 +1,59 @@
+import { BaseService } from '../services/base-service';
+
+export const detectPacFiles = async (gitService: BaseService) => {
+  let pacFiles = [];
+  const isTektonFolderPresent = await gitService.isTektonFolderPresent();
+
+  if (isTektonFolderPresent) {
+    pacFiles.push('.tekton');
+    const { files: pipelineFiles } = await gitService.getRepoFileList({
+      specificPath: '.tekton',
+      includeFolder: true,
+    });
+    if (pipelineFiles.length > 0) {
+      pacFiles = [...pacFiles, ...pipelineFiles];
+    }
+  }
+  return pacFiles;
+};
+
+export const verifyPipelinesFile = async (file: string, gitService: BaseService) => {
+  const trimmedFile = file?.startsWith('.tekton/') ? file?.replace('.tekton/', '') : file;
+
+  const resourceContent = await gitService.getFileContent(`.tekton/${trimmedFile}`);
+  return resourceContent?.split(/\r?\n/).includes('kind: PipelineRun');
+};
+
+export const isPipelineFilePresent = async (
+  pacFiles: string[],
+  gitService: BaseService,
+): Promise<boolean> => {
+  if (!pacFiles || pacFiles?.length === 0) {
+    return false;
+  }
+  const pipelineFile: string = pacFiles.find(
+    (file) => file?.endsWith('.yaml') || file?.endsWith('.yml'),
+  );
+
+  const isKindPipelineRunPresent = await verifyPipelinesFile(pipelineFile, gitService);
```

Comment:
The new implementation checks just the first yaml file, is this fine @Lucifergene @divyanshiGupta?

Alternatively you can do something like:

```typescript
for await (const pacFile of pacFiles) {
  if (pacFile.endsWith('.yaml') || pacFile.endsWith('.yml')) {
    if (await verifyPipelinesFile(pipelineFile, gitService)) {
      return true;
    }
  }
}
return false;
```

Wdyt? Disclaimer: I don't read the .tekton spec to understand if its allowed to put there multiple yaml files.

Reply:
Updated :+1: